### PR TITLE
Upgrade Java and Python Integration Example

### PR DESCRIPTION
**Pull request by LesiaChaban**

### CODE REVIEW

File: .github/workflows/graalpy-notebook-example.yml [hunk 1]
```diff
@@ -20,21 +20,19 @@ jobs:
...
-            java-version: '17'
+        java-version: ['21']
     steps:
       - uses: actions/checkout@v3
       - uses: graalvm/setup-graalvm@v1
         with:
-          version: ${{ matrix.version }}
           java-version: ${{ matrix.java-version }}
+          distribution: 'graalvm'
           components: 'python'
```
Comment (olyagpl) [inline .github/workflows/graalpy-notebook-example.yml:30]: You can't install 'python' component on GraalVM for JDK 21. We should keep this demo testing on java-version: '17'.

File: graalpy-notebook-example/pom.xml [hunk 2]
```diff
@@ -128,9 +144,8 @@
                         <configuration>
                             <rules>
                                 <requireFilesExist>
-                                    <message>You must set JAVA_HOME to a GraalVM with Python installed.</message>
                                     <files>
-                                        <file>${env.JAVA_HOME}/bin/graalpy</file>
```
Comment (olyagpl) [inline graalpy-notebook-example/pom.xml:133]: Why do we need to change the env name?

File: graalpy-notebook-example/README.md [hunk 3]
```diff
@@ -4,16 +4,27 @@ This example demonstrates how to integrate Python in a Java application and run
...
 ## Preparation
 
-1. Download and install the latest GraalVM JDK with the Python support using the [GraalVM JDK Downloader](https://github.com/graalvm/graalvm-jdk-downloader). 
+1. Download and install the latest GraalVM JDK using [SDKMAN!](https://sdkman.io/).
+    ```bash
+    sdk install java 21.0.1-graal
+    ```
+2. Download and install  GraalPy using `pyenv`:
+    ```bash
+    pyenv install graalpy-23.1.0
```
Comment (olyagpl) [inline graalpy-notebook-example/README.md:20]: It's already `graalpy-23.1.1`

File: graalpy-notebook-example/pom.xml (hunk 2 above)
Comment (LesiaChaban) [reply graalpy-notebook-example/pom.xml:133]: It doesn’t change the `env` name. The name of env is here `<argument>venv</argument>`.

File: .github/workflows/graalpy-notebook-example.yml (hunk 1 above)
Comment (olyagpl) [reply .github/workflows/graalpy-notebook-example.yml:30]: That's one option. As I understand, you need to install GraalPy JVM standalone 23.1.1 and export GRAALPY_HOME to to using GitHub actions workflow.

Comment (fniephaus) [reply .github/workflows/graalpy-notebook-example.yml:30]: If we decide to update this demo to a newer version of GraalPy, we no longer can/should use the `setup-graalvm` action, but you may be able to use `setup-python`, which [supports GraalPy](https://github.com/actions/setup-python/pull/694).

**Review by timfel — COMMENTED** [review]
I'll check out this PR locally and fix it up

File: graalpy-notebook-example/pom.xml
[the remark this answers is not in the document]
Comment (olyagpl) [reply graalpy-notebook-example/pom.xml:1]: Are we going to publish our graalpy-maven-plugin? That's great news if so!

[the remark this answers is not in the document]
Comment (timfel) [reply graalpy-notebook-example/pom.xml:1]: yes, it goes out with the next release

[the remark this answers is not in the document]
Comment (LesiaChaban) [reply graalpy-notebook-example/pom.xml:1]: So @olyagpl we wait and refactor the Maven plugin for GraalPy (when it comes out)? Leave GitHub workflow running with JDK 17?

**Pull request closed without merging**